src/sortStudents.ts

```diff
@@ -1,16 +1,74 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function avagareGrade(student: Student) : number {
+  return student.grades.reduce((a, b) => a + b, 0) / student.grades.length;
+}
+
+export function sortStudents(
+  students: Student [],
+  sortBy: SortType,
+  order: SortOrder,
+) : Student [] {
+  const studentCopy = [...students];
+
+  studentCopy.sort((firstStudent, secondStudent) => {
+    switch (sortBy) {
+      case SortType.Name:
+        if (order === 'asc') {
+          return firstStudent.name.localeCompare(secondStudent.name);
+        }
+
+        return secondStudent.name.localeCompare(firstStudent.name);
+
+      case SortType.Surname:
+        if (order === 'asc') {
+          return firstStudent.surname.localeCompare(secondStudent.surname);
+        }
+
+        return secondStudent.surname.localeCompare(firstStudent.surname);
```

Comment:
combine similar data to one case
```suggestion
      case SortType.Name:
      case SortType.Surname:
        if (order === 'asc') {
          return firstStudent[sortBy].localeCompare(secondStudent[sortBy]);
        }

        return secondStudent[sortBy].localeCompare(firstStudent[sortBy]);
```